unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Loader/WearableLoader.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            this.wearable = wearable;
+            this.retriever = retriever;
+        }
+
+        public async UniTask Load(GameObject container, AvatarSettings settings, CancellationToken ct = default)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            try
+            {
+                bool bodyshapeDirty = currentSettings.bodyshapeId != settings.bodyshapeId;
+                currentSettings = settings;
+                if (status == IWearableLoader.Status.Succeeded && !bodyshapeDirty)
+                {
+                    AvatarSystemUtils.PrepareMaterialColors(rendereable, currentSettings.skinColor, currentSettings.hairColor);
+                    return;
+                }
+
+                retriever.Dispose();
+
+                try
+                {
+                    await LoadWearable(container, wearable, settings.bodyshapeId, ct);
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception e)
+                {
+                    // Ignored so we try to fallback (if needed)
+                }
+
+                // Succeeded
+                if (rendereable != null)
+                {
+                    AvatarSystemUtils.PrepareMaterialColors(rendereable, currentSettings.skinColor, currentSettings.hairColor);
+                    status = IWearableLoader.Status.Succeeded;
+                    return;
+                }
+
+                //Try getting a default if category is needed
+                if (AvatarSystemUtils.IsCategoryRequired(wearable.data.category))
+                    await FallbackToDefault(container, ct);
+
+                if (rendereable != null)
+                {
+                    AvatarSystemUtils.PrepareMaterialColors(rendereable, currentSettings.skinColor, currentSettings.hairColor);
+                    status = IWearableLoader.Status.Defaulted;
+                }
+                else
+                    status = IWearableLoader.Status.Failed;
+            }
+            catch (OperationCanceledException)
+            {
+                Dispose();
+                throw;
+            }
+        }
+
+        private async UniTask FallbackToDefault(GameObject container, CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+
+            try
+            {
+                string wearableId = WearableLiterals.DefaultWearables.GetDefaultWearable(currentSettings.bodyshapeId, wearable.data.category);
+                Debug.Log($"Falling back {wearable.id} to wearable {wearableId}");
+
+                WearableItem defaultWearable = await defaultWearablesResolver.Resolve(wearableId, ct);
+
+                await LoadWearable(container, defaultWearable, currentSettings.bodyshapeId, ct);
+            }
+            catch (OperationCanceledException)
+            {
+                //No disposing required
```

Comment:
unnecessary comment

Reply:
Same as before, I rather leave them explicitly so they can be used as a template when working with UniTask